ftwhttp/client_test.go

```diff
@@ -1,63 +1,109 @@
 package ftwhttp
 
 import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
 	"testing"
 
-	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
 )
 
-func TestNewClient(t *testing.T) {
-	c, err := NewClient(NewClientConfig())
-	assert.NoError(t, err)
-
-	assert.NotNil(t, c.Jar, "Error creating Client")
+type clientTestSuite struct {
+	suite.Suite
+	client *Client
+	ts     *httptest.Server
 }
 
-func TestConnectDestinationHTTPS(t *testing.T) {
-	d := &Destination{
-		DestAddr: "example.com",
-		Port:     443,
-		Protocol: "https",
-	}
-
-	c, err := NewClient(NewClientConfig())
-	assert.NoError(t, err)
+func TestClientTestSuite(t *testing.T) {
+	suite.Run(t, new(clientTestSuite))
+}
 
-	err = c.NewConnection(*d)
-	assert.NoError(t, err, "This should not error")
-	assert.Equal(t, "https", c.Transport.protocol, "Error connecting to example.com using https")
+func (s *clientTestSuite) SetupTest() {
+	var err error
+	s.client, err = NewClient(NewClientConfig())
+	s.NoError(err)
+	s.Nil(s.client.Transport, "Transport not expected to initialized yet")
 }
 
-func TestDoRequest(t *testing.T) {
-	d := &Destination{
-		DestAddr: "httpbin.org",
-		Port:     443,
-		Protocol: "https",
+func (s *clientTestSuite) TearDownTest() {
+	if s.ts != nil {
+		s.ts.Close()
 	}
+}
 
-	c, err := NewClient(NewClientConfig())
-	assert.NoError(t, err)
+func (s *clientTestSuite) httpTestServer() {
+	s.ts = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		resp := new(bytes.Buffer)
+		for key, value := range r.Header {
+			_, err := fmt.Fprintf(resp, "%s=%s,", key, value)
+			s.NoError(err)
+		}
+
+		_, err := w.Write(resp.Bytes())
+		s.NoError(err)
+	}))
+}
 
-	req := generateBaseRequestForTesting()
+func (s *clientTestSuite) httpsTestServer() {
```

Comment:
Use one parameterized method instead of two with duplicated bodies?

Reply:
Sure.